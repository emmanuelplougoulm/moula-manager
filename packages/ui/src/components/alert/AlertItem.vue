<template>
  <p class="text-default-regular">
    {{ text }}
  </p>
</template>

<script setup lang="ts">
import { computed } from 'vue';
import { colorTypeValidator } from './validator';

const props = defineProps({
  text: {
    type: String,
    required: true
  },
  colorType: {
    type: String,
    default: 'primary',
    validator: colorTypeValidator
  }
});

const color = computed(() => `var(--${props.colorType}-30)`);
const backgroundColor = computed(() => `var(--${props.colorType}-90)`);
</script>

<style scoped>
p {
  padding: 0.75rem 1.25rem;
  border-radius: 0.5rem;
  line-height: 1.188rem;
  color: v-bind(color);
  background-color: v-bind(backgroundColor);
  margin: 0;
}
</style>
