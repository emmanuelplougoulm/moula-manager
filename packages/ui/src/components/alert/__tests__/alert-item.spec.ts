import { mount } from '@vue/test-utils';
import AlertItem from '../AlertItem.vue';

describe('AlertItem tests', () => {
  it('should display the component with its text and colorType props', () => {
    const wrapper = mount(AlertItem, {
      props: {
        text: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Et penatibus eleifend vulputate condimentum tempus.',
        colorType: 'warning'
      }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
