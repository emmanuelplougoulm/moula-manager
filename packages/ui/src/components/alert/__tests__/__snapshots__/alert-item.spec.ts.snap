// Vitest Snapshot v1

exports[`AlertItem tests > should display the component with its text and colorType props 1`] = `"<p data-v-a2a80cda=\\"\\" class=\\"text-default-regular\\">Lorem ipsum dolor sit amet, consectetur adipiscing elit. Et penatibus eleifend vulputate condimentum tempus.</p>"`;
